Desktop/components/JASP/Widgets/MainPage.qml

```diff
@@ -318,7 +318,8 @@ Item
 			{
 				resultsJsInterface.resultsLoaded = loadRequest.status === WebEngineView.LoadSucceededStatus;
 				setTranslatedResultsString();
-				runJavaScript(`window.sendUrlWhitelist(${JSON.stringify(urlWhitelist)})`); //sent urlWhitelist to js side
+				if(resultsJsInterface.resultsLoaded)
```

Comment:
I understand you add this condition, but did you get a situation when this condition was false?

Reply:
I never, the problem is `window.sendUrlWhitelist` from qml side always runs before defined in js side right, so qml will warning `window.sendUrlWhitelist is not a function`.

if I understand right if `WebEngineView` not loaded then browser will never be opened so I suppose `resultsLoaded` will also not loaded.

Reply:
I was afraid that window.sendUrlWhitelist was then never called. But apparently the onLoadingChanged is called 2 times: once when the resultsJsInterface.resultsLoaded is false, and once when it is true. So your code is perfect!